### PR TITLE
Columns block styling

### LINKED ISSUE #49: style columns block
Covers styling two column blocks as below:

Screenshot 2023-02-27 at 4 31 23 PM

Screenshot 2023-02-27 at 4 31 37 PM

### PR DESCRIPTION
**Pull request by asthabh23**
Fix #49 

Test URLs:
- Before: https://main--delta--hlxsites.hlx.page/us/en/skymiles/how-to-earn-miles/overview
- After: https://issue-49--delta--hlxsites.hlx.page/us/en/skymiles/how-to-earn-miles/overview

### CODE REVIEW

**Comment by ramboz** [comment]
Looking closer at the document and PR:
1. I don't think using `h3` is really semantically correct in the 2nd columns group since those labels essentially just serve a descriptive role for the image above. I think these probably should be figures like in the first group
2. Comparing the styles with the official website, it seems we still need to work on those a bit. I'm fine with minor deviations, and improvements on the existing layout if needed, but we should try to stick to something close in any case.

Maybe this is still WIP, so I might be anticipating a bit here.

|Breakpoint|delta.com|Franklin|
|-|-|-|
|375px|Screenshot 2023-02-27 at 3 01 56 PM|Screenshot 2023-02-27 at 3 02 03 PM
|768px|Screenshot 2023-02-27 at 3 04 25 PM|Screenshot 2023-02-27 at 3 04 35 PM
|992px|Screenshot 2023-02-27 at 3 09 36 PM|Screenshot 2023-02-27 at 3 09 43 PM

File: us/en/skymiles/scripts/scripts.js [hunk 1]
```diff
@@ -60,7 +60,7 @@ function decorateHyperlinkImages(container) {
 
 function decorateReferences(container) {
   const REFERENCE_TOKENS = /(\*+|[†‡])/g;
-  [...container.querySelectorAll('p,a,li')]
+  [...container.querySelectorAll('p,a,li,h3')]
```
[the remark this answers is not in the document]
Comment (asthabh23) [reply us/en/skymiles/scripts/scripts.js]: Incorporated in commit [ee027ea](https://github.com/hlxsites/delta/pull/52/commits/ee027eafa9459cbee4e92f3412aafdc11e397f1c)

File: us/en/skymiles/blocks/columns/columns.css [hunk 2]
```diff
@@ -4,12 +4,34 @@
...
 
-.columns img {
+.columns p a img {
     width: auto;
-    height: 75px;
+    height: auto;
 }
 
-@media (min-width: 900px) {
+.columns > div > div > p > a > figure > figcaption {
```
[the remark this answers is not in the document]
Comment (asthabh23) [reply us/en/skymiles/blocks/columns/columns.css:12]: No, I tried using `.columns a figcaption` but that doesn't help, styles don't reflect as expected

File: us/en/skymiles/blocks/columns/columns.css [hunk 3]
```diff
@@ -4,12 +4,34 @@
...
+    font-size: 17px;
+    padding-top: 30px;
+    padding-bottom: 30px;
+    width: 200px;
+    margin: 0 auto;
+}
+
+.columns > div > div > h3 {
```
[the remark this answers is not in the document]
Comment (asthabh23) [reply us/en/skymiles/blocks/columns/columns.css]: Incorporated in commit [ee027ea](https://github.com/hlxsites/delta/pull/52/commits/ee027eafa9459cbee4e92f3412aafdc11e397f1c)

File: us/en/skymiles/blocks/columns/columns.css [hunk 4]
```diff
@@ -4,12 +4,34 @@
...
+    font-size: 17px;
+    width: 280px;
+    margin: 0 auto;
+    margin-bottom: 50px;
+    line-height: 25px;
+    margin-top: 30px;
+    font-weight: 300;
+}
+
+.columns-container .columns-wrapper {
```
[the remark this answers is not in the document]
Comment (asthabh23) [reply us/en/skymiles/blocks/columns/columns.css]: Incorporated in commit [ee027ea](https://github.com/hlxsites/delta/pull/52/commits/ee027eafa9459cbee4e92f3412aafdc11e397f1c)

**Comment by asthabh23** [comment]
@ramboz For your above comment, I am unable to follow point 1.
For point 2, I differ a bit in the approach taken, I also see that the screenshots that you have shared for current implementation don't actually match with what I see on my branch [issue-49](https://issue-49--delta--hlxsites.hlx.page/us/en/skymiles/how-to-earn-miles/overview)
 (the screenshots in your comment above seem to be of [main](https://main--delta--hlxsites.hlx.page/us/en/skymiles/how-to-earn-miles/overview) where changes are not merged yet. Could you please check that again? I am still sharing a comparison of what I see as below:


Breakpoint | delta.com | Franklin | Comparison
-- | -- | -- | --
375px | Screenshot 2023-02-28 at 12 13 32 PM | Screenshot 2023-02-28 at 12 14 08 PM | I feel the original site has unnecessary empty spacing around content, I have tried to make it visually better utilising the spacing well. Also the first image on the original site is blurry, which looks better in our version.
768px |Screenshot 2023-02-28 at 12 18 44 PM |Screenshot 2023-02-28 at 12 19 37 PM | Same as above comment, our implementation would look better considering image size consistencies as well.
992px | Screenshot 2023-02-28 at 12 22 44 PM | Screenshot 2023-02-28 at 12 23 20 PM | Same consideration for image consistencies and utilising empty space as well.


Also I have a point regarding breakpoint at 375px, in Franklin we generally don't prefer putting breakpoint below 600px (anything below that is considered as part of the default mobile view and any break there should be avoided), do we still prefer to add this break for this project?

**Comment by ramboz** [comment]
@asthabh23 Agree with all your points above! And the updated version definitely looks better (both compared to the original website and to the 1st version of this PR) 👏🏼 

As to the breakpoints, I'm not sure about that `600px` rule. There are still a lot of old mobile devices that don't have a good pixel density and where I think it still makes sense to separate. Ultimately I also think this is a designing and marketing decision at the project level, and since they have the breakpoint in their current fresh air syle guide, I'd leave it up to them to revise that in the next major iteration of their website

**Review by ramboz — APPROVED** [review]

**Comment by ramboz** [comment]
@asthabh23 I went ahead and rebased this on the latest update of the main branch and merged. Happy to revisit when you are back if needed.

**Comment by asthabh23** [comment]
@ramboz , I see the images size isn't the same for second columns block for smaller view ports. Do you see the same at your end? I feel there would be a need to revisit the changes to fix the alignments

**Comment by ramboz** [comment]
Yes, you're right. I think there are a few elements we need to revise:
1. they seem to be using responsive images also in the cards, so they don't use the same image for mobile and desktop and our current logic only takes one of the 2… so this creates some differences
2. the decoration logic for the `figure` element isn't really flexible, and the 2nd block didn't actually trigger it (because we it expects a link that isn't present), so the markup ends up being different.

**Comment by ramboz** [comment]
Filed https://github.com/hlxsites/delta/issues/76 to track this

**Pull request merged**